src/tsdb.c

```diff
@@ -604,4 +605,37 @@ int SeriesCalcRange(Series *series,
 
 timestamp_t CalcWindowStart(timestamp_t timestamp, size_t window) {
     return timestamp - (timestamp % window);
+}
+
+timestamp_t SkipExpiredSamples(Series *series, long long *skipped) {
```

Comment:
The name is a bit confusing.

Reply:
Agree.
How about `getFirstValidTimestamp`?
Any other idea?

Reply:
@ashtul Fine by me, just a add a comment to explain that valid is meant to be "in-range" because we don't have a notion of valid/invalid samples.